### PR TITLE
SI-6795 Adds negative check for "abstract override" on types in traits

### PR DESCRIPTION
**Pull request by JamesIry**
"abstract override" shouldn't was being allowed on types in traits but the result
made no sense and the spec says that shouldn't be allowed.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala [hunk 1]
```diff
@@ -1077,7 +1077,7 @@ trait ContextErrors {
             "`override' modifier not allowed for constructors"
 
           case AbstractOverride =>
-            "`abstract override' modifier only allowed for members of traits"
+            "`abstract override' modifier only allowed for non-type members of traits"
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala]: I'm afraid you should call them "value members".

Comment (JamesIry) [inline src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala]: We'll have two distinct errors: the old one for "only allowed for members of traits" and a new one for "not allowed on type members"

**Comment by paulp** [comment]
lgtm.

**Comment by paulp** [comment]
"Failed" is random-build-abortese for "Succeeded."

```
docs.jline:
    [mkdir] Created dir: /localhome/jenkins/a/workspace/pr-scala-testsuite-linux-opt/build/scaladoc/jline
 [scaladoc] Documenting 34 source files to /localhome/jenkins/a/workspace/pr-scala-testsuite-linux-opt/build/scaladoc/jline
 [scaladoc] model contains 38 documentable templates
Build was aborted
Archiving artifacts
Finished: ABORTED
```

**Comment by JamesIry** [comment]
PLS REBUILD pr-scala-testsuite-linux-opt

**Comment by JamesIry** [comment]
That was a mistake. Deleted the wrong branch.

**Pull request closed without merging**